Batteries/Data/Vector/Monadic.lean

```diff
@@ -0,0 +1,170 @@
+/-
+Copyright (c) 2024 Shreyas Srinivas. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shreyas Srinivas
+-/
+
+import Batteries.Data.Vector.Basic
+import Init.Data.Array.Basic
+import Init.Data.Array.Lemmas
+import Batteries.Classes.SatisfiesM
+
+/-!
+## Monadic Definitions for Vectors
+In this file we add monadic definitions for statically sized vectors.
+-/
+
+namespace Batteries
+
+namespace Vector
+
+/--
+`modifyM v i f` applies a monadic transformation `f` to `v[i]`
+-/
+def modifyM [Monad m] (v : Vector α n) (i : Nat)
+  (f : α → m α) : m (Vector α n) := do
+    return ⟨←Array.modifyM v.toArray i f,
+      by rw[←v.size_eq]; sorry⟩
```

Comment:
Any help in fixing this `sorry` is welcome. I found a lemma by the name `Array.size_modifyM` which however is not a straightforward equality.